examples/oop/data_driven_physics.cpp

```diff
@@ -0,0 +1,149 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+
+// In this example we will have a look at how many utilities from eve
+// come together in writing a non-trivial data oriented computation
+//
+// We have a set of balls at some altitude z in a gravity field G.
+// At altitude 0, lays a floor on which each balls will bounce with some restitution coefficient
+// The following code run a discrete simulation of this experiment using SoA vector and algorihms
+
+#include <eve/algo/any_of.hpp>
+#include <eve/algo/container/soa_vector.hpp>
+#include <eve/algo/transform.hpp>
+#include <eve/product_type.hpp>
+#include <eve/eve.hpp>
+
+#include <chrono>
+#include "display.hpp"
+#include "test.hpp"
+
+// Ball entity -------------------------------------------------------------------------------------
+struct ball : eve::struct_support < ball
+                                  , float, float, float, std::int32_t
+                                  >
+{
+  EVE_FORCEINLINE friend decltype(auto) position(eve::like<ball> auto &&self)
+  {
+    return get<0>(std::forward<decltype(self)>(self));
+  }
+
+  EVE_FORCEINLINE friend decltype(auto) speed(eve::like<ball> auto &&self)
+  {
+    return get<1>(std::forward<decltype(self)>(self));
+  }
+
+  EVE_FORCEINLINE friend decltype(auto) elasticity(eve::like<ball> auto &&self)
+  {
+    return get<2>(std::forward<decltype(self)>(self));
+  }
+
+  EVE_FORCEINLINE friend decltype(auto) count(eve::like<ball> auto &&self)
+  {
+    return get<3>(std::forward<decltype(self)>(self));
+  }
+};
+
+// Simulation step ---------------------------------
+struct update
+{
+  auto operator()(eve::like<ball> auto const& b) const noexcept
+  {
+    auto falling = b;
+    auto bounced = b;
+
+    // Compute the case of the ball not bouncing
+    // Update position
+    position(falling) += speed(falling)*dt;
+
+    // Update velocity with gravity
+    speed(falling)    -= g*dt;
+
+    // Compute the bounced ball case
+    speed(bounced)    *= -elasticity(bounced);
+    count(bounced)++;
+
+    // Select the proper one based on position
+    return eve::if_else(  count(b) < max_bounce
```

Comment:
I don't get this `count` and `max_bounce`.
The speed goes down, right?
So, maybe, just let it go down? and trim below a treshold?

Reply:
Whenever the position reaches 0, we bounce by reverting the speed vector sign. It'll then get affected by gravity again, then rebound, etc... It will bounce until the speed is close to 0, at which point we stick the particle to the ground.

the count is used to count how many bounces happened for a particule and we stop after a max number of bounces.

Reply:
Why do we stop after max bounce? It should stop itself, right?
I mean, you can say if the 'speed < treshold`, speed = 0. That's it.